pkgs/tools/graphics/grace/default.nix

```diff
@@ -0,0 +1,74 @@
+{ lib
+, gccStdenv
+, fetchurl
+, coreutils
+, motif
+, fftw
+, netcdf
+, libpng
+, libjpeg_original
+, libX11
+, libSM
+, libXext
+, libXmu
+, libXp
+, libXpm
+, libXt
+}:
+
+gccStdenv.mkDerivation rec {
+    pname = "grace";
+    version = "5.1.25";
+    src = fetchurl {
+        url = "ftp://ftp.fu-berlin.de/unix/graphics/grace/src/grace5/grace-5.1.25.tar.gz";
+        sha256 = "751ab9917ed0f6232073c193aba74046037e185d73b77bab0f5af3e3ff1da2ac";
+    };
+
+    buildInputs = [
+      coreutils
+      motif
+      fftw
+      netcdf
+      libpng
+      libjpeg_original
+      libX11
+      libSM
+      libXext
+      libXmu
+      libXp
+      libXpm
+      libXt
+    ];
+
+    configurePhase = ''
+        ./configure --enable-grace-home=$out \
+                    --prefix=$out \
+                    --disable-debug \
+                    --with-bundled-t1lib=yes
+    '';
+
+    buildPhase = ''
+        NIX_CFLAGS_COMPILE="-Wno-implicit-function-declaration -Wno-format-security $NIX_CFLAGS_COMPILE"
```

Comment:
Why is this needed. If they're really needed then this isnt the correct way to declare them

Reply:
-Wno-implicit-function-declaration is indeed not necessary.

-Wno-format-security is required for a successful built. Grace does not implement format strings correctly. Example error message:

objects.c:960:22: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]

Reply:
Performing further error checking, we can ignore -Wno-implicit-function-declaration. The crashing binary is only produced if clang is used.  But the -Wno-format-security is still necessary for a successful build due to the format strings. I deleted the build phase overwrite but what is the preferred way of setting of setting cflags?

Now I use:
`NIX_CFLAGS_COMPILE=-Wno-format-security `

Thank you for your patience.

Reply:
```
NIX_CFLAGS_COMPILE = [ "-Wno-format-security" ] 
  ++ lib.optionals stdenv.cc.isClang
    [ "-Wno-implicit-function-declaration" ];
```

Don't put this in any phase